From 48a5bfedb632e57db58e52ac7b6172f6efd3387e Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Mon, 7 Mar 2022 12:06:16 +0100
Subject: [PATCH 090/106] Revert "bfq: optimize the calculation of
 bfq_weight_to_ioprio()"

This reverts commit ce85df3f80ecc723269f9b9fa87f0503a63abaff.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/bfq-wf2q.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/block/bfq-wf2q.c b/block/bfq-wf2q.c
index b5ea8bf74..b52da3846 100644
--- a/block/bfq-wf2q.c
+++ b/block/bfq-wf2q.c
@@ -518,9 +518,8 @@ unsigned short bfq_ioprio_to_weight(int ioprio)
  */
 static unsigned short bfq_weight_to_ioprio(int weight)
 {
-	int ioprio = IOPRIO_NR_LEVELS  - weight / BFQ_WEIGHT_CONVERSION_COEFF;
-
-	return ioprio < 0 ? 0 : min_t(int, ioprio, IOPRIO_NR_LEVELS - 1);
+	return max_t(int, 0,
+		     IOPRIO_NR_LEVELS * BFQ_WEIGHT_CONVERSION_COEFF - weight);
 }
 
 static void bfq_get_entity(struct bfq_entity *entity)
-- 
2.35.1.354.g715d08a9e5

